website/server/controllers/api-v3/auth.js

```diff
@@ -146,6 +146,35 @@ api.loginSocial = {
   },
 };
 
+// Called by apple for web authentication.
+api.redirectApple = {
+  method: 'POST',
+  middlewares: [authWithHeaders({
+    optional: true,
+  })],
+  url: '/user/auth/apple',
+  async handler (req, res) {
+    if (req.body.id_token) {
+      req.body.network = 'apple';
+      return loginSocial(req, res);
+    }
+    return res.redirect(303, `/static/apple-redirect?code=${req.body.code}`);
+  },
+};
+
+// Called as a callback by Apple. Internal route
+api.loginApple = {
+  method: 'GET',
+  middlewares: [authWithHeaders({
+    optional: true,
+  })],
+  url: '/user/auth/apple',
+  async handler (req, res) {
+    req.body.network = 'apple';
+    return loginSocial(req, res);
```

Comment:
So, here, `loginSocial` with `req.body.network = 'apple';` calls `_appleProfile` which seems to need either `req.body.code` or `req.body.id_token` but this one being a GET request it'll never have a body?


```

  if (req.body.code) {
    const response = await auth.accessToken(req.body.code);
    idToken = jwt.decode(response.id_token);
  } else if (req.body.id_token) {
    idToken = await jwt.verify(req.body.id_token, applePublicKey, { algorithms: ['RS256'] });
  } 
```

Reply:
You're right. But it does work 🤔 . Does express maybe put the query arguments to also be available through `req.body`?

Reply:
https://www.npmjs.com/package/body-parser I think the query params are available in `req.body` because of this package that we use.

Reply:
hmm weird, that should only fill `req.body` with data from the body. Anyway, just to be clearer, could we change the code to use `req.query` to specify where it's coming from?

Reply:
we _do_ need query for handling iOS login unfortunately. But I'll change it, so that it checks both of those values.